### DISCUSSION TITLE
Video Node with Chroma Key using Augmented Image Detection

### DISCUSSION THREAD

**Discussion by dhavalenstead** [sceneview/sceneview#403] [General]
Want to create logic that will detect image and play video with chroma key color. Is there any example.  I seen video node but its commented

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, you can combine Augmented Images with VideoNode. For chroma key (green screen removal), you'd need a custom Filament material:

1. **Detect image** via ARCore Augmented Images (see #284)
2. **Place VideoNode** at the detected image anchor
3. **Chroma key** requires a custom material shader that discards green pixels

SceneView doesn't have a built-in chroma key material, but you can create a custom Filament material (.mat file) with a fragment shader that keys out a specific color, then apply it to your VideoNode. The Filament material system supports custom shaders for this kind of effect.
